### ISSUE TITLE
RPM file associations not working.

### ISSUE THREAD

**Issue by schadfield** [fvarrui/JavaPackager#316]
I tested the Shogi Explorer RPM on RHEL 9 and the link is created in /usr/share/mime/packages but the file ShogiExplorer.xml does not exist in /opt/ShogiExplorer.

Using version 1.7.0 with:

https://github.com/schadfield/shogi-explorer

**Comment by fvarrui** [comment]
Hi @schadfield!
Oh! 😞  I thought it was working fine as I tried on Fedora. I'm going to test it and tell you something ASAP.
Thanks!!

**Comment by fvarrui** [comment]
Hi @schadfield!
I've made some improvements on RPM generation. You can find those changes on `issue-316` branch. I've also uploaded a SNAPSHOT version: `1.7.3-20230507.163637-2` (if you have any doubt about how to use a snapshot, take a look to this [link](https://github.com/fvarrui/JavaPackager#how-to-use-snapshot-versions)). 
Please, test it and give me some feedback. 
Thanks!

**Comment by fvarrui** [comment]
hi @schadfield!
Any news?

**Comment by schadfield** [comment]
Sorry, I haven't been able to work on it recently.

**Comment by schadfield** [comment]
I am afraid I cannot get the snapshot version to compile.

**Comment by fvarrui** [comment]
Don't worry, I'm publishing snapshot versions on Nexus repo, so you don't have to build the plugin from sources. Just follow instructions in this [link](https://github.com/fvarrui/JavaPackager#how-to-use-snapshot-versions).

**Comment by schadfield** [comment]
That is what I cannot get to work.

**Comment by schadfield** [comment]
I have managed to get further by removing spaces from the <name> sections of the plugin configuration. 1.7.3-SNAPSHOT is (mostly) building now.

The problem is still the same: the file "/opt/ShogiExplorer/ShogiExplorer.xml" is missing from the RPM file.

In addition the AppImage fails:

 /bin/sh: line 1: ARCH=x86_64: command not found
    AppImage generation failed due to: Command execution failed: ARCH=x86_64 /home/stephen/Projects/shogi-explorer/target/packages/linux/assets/appimagetool --appimage-extract-and-run /home/stephen/Projects/shogi-explorer/target/packages/linux/assets/AppDir /home/stephen/Projects/shogi-explorer/target/packages/linux/ShogiExplorer.AppImage

**Comment by fvarrui** [comment]
If you use 1.7.3-SNAPSHOT, Maven will take the latest version, built for another patch, not the version including the patch you need.

**Comment by schadfield** [comment]
Plugin io.github.fvarrui:javapackager:1.7.3-20230507.163637-2 or one of its dependencies could not be resolved: The following artifacts could not be resolved: io.github.fvarrui:javapackager:jar:1.7.3-20230507.163637-2 (absent): Could not find artifact io.github.fvarrui:javapackager:jar:1.7.3-20230507.163637-2 in nexus (https://oss.sonatype.org/content/repositories/snapshots) -> [Help 1]

**Comment by fvarrui** [comment]
Oh! 😮 sorry, it seems that old jars are removed from Nexus snapshot repo. I've just checked that this version is no longer available. I've just published a new version with your patch: `1.7.3-20230717.142757-16`.

I've just tried and this is the content of the generated RPM:

![image](https://github.com/fvarrui/JavaPackager/assets/4354486/fced8864-bd8d-4eae-a067-3c9a1cc4f9a6)

I've made some changes to reduce your JP configuration ... this is the `pom.xml` I used to build your project:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.chadfield</groupId>
    <artifactId>ShogiExplorer</artifactId>
    <version>4.18.1</version>
    <packaging>jar</packaging>
    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <maven.compiler.source>17</maven.compiler.source>
        <maven.compiler.target>17</maven.compiler.target>
        <exec.mainClass>com.chadfield.shogiexplorer.ShogiExplorer</exec.mainClass>
    </properties>
    <name>ShogiExplorer</name>
    <dependencies>
        <dependency>
            <groupId>com.thoughtworks.xstream</groupId>
            <artifactId>xstream</artifactId>
            <version>1.4.20</version>
        </dependency>
        <dependency>
            <groupId>org.netbeans.external</groupId>
            <artifactId>AbsoluteLayout</artifactId>
            <version>RELEASE170</version>
        </dependency>
        <dependency>
            <groupId>com.ibm.icu</groupId>
            <artifactId>icu4j</artifactId>
            <version>72.1</version>
        </dependency>
        <dependency>
            <groupId>org.jfree</groupId>
            <artifactId>jfreechart-swt</artifactId>
            <version>1.0</version>
        </dependency>
        <dependency>
            <groupId>net.harawata</groupId>
            <artifactId>appdirs</artifactId>
            <version>1.2.1</version>
        </dependency>
        <dependency>
            <groupId>org.apache.xmlgraphics</groupId>
            <artifactId>batik-transcoder</artifactId>
            <version>1.16</version>
        </dependency>
        <dependency>
            <groupId>org.apache.xmlgraphics</groupId>
            <artifactId>batik-codec</artifactId>
            <version>1.16</version>
        </dependency>
    </dependencies>
    <build>
        <resources>
            <resource>
                <directory>src/main/resources</directory>
                <filtering>true</filtering>
            </resource>
        </resources>
        <plugins>
            <plugin>
                <artifactId>maven-assembly-plugin</artifactId>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <goals>
                            <goal>single</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <archive>
                        <manifest>
                            <addClasspath>true</addClasspath>
                            <mainClass>${exec.mainClass}</mainClass>
                        </manifest>
                    </archive>
                    <descriptorRefs>
                        <descriptorRef>jar-with-dependencies</descriptorRef>
                    </descriptorRefs>
                </configuration>
            </plugin> 
            <plugin>
                <groupId>io.github.fvarrui</groupId>
                <artifactId>javapackager</artifactId>
                <version>1.7.3-20230717.142757-16</version>
                <configuration>
                    <mainClass>${exec.mainClass}</mainClass>
                    <bundleJre>false</bundleJre>
                    <url>https://github.com/schadfield/shogi-explorer</url>
                    <displayName>Shogi Explorer</displayName>
                    <jreMinVersion>17</jreMinVersion>
                    <fileAssociations>
                        <fileAssociation>
                            <description>Shogi KIFU</description>
                            <extension>kif</extension>
                            <mimeType>application/x-shogi-explorer</mimeType>
                        </fileAssociation>
                    </fileAssociations>
                </configuration>
                <executions>
                    <execution>
                        <id>bundling-for-windows</id>
                        <phase>none</phase>
                        <goals>
                            <goal>package</goal>
                        </goals>
                        <configuration>
                            <generateInstaller>false</generateInstaller>
                            <createZipball>true</createZipball>
                            <platform>windows</platform>
                            <name>Shogi Explorer</name>
                            <outputDirectory>target/packages/windows</outputDirectory>
                            <vmArgs>
                                <vmArg>-Dfile.encoding=UTF-8</vmArg>
                            </vmArgs>
                        </configuration>
                    </execution>
                    <execution>
                        <id>bundling-for-mac</id>
                        <phase>none</phase>
                        <goals>
                            <goal>package</goal>
                        </goals>
                        <configuration>
                            <platform>mac</platform>
                            <name>Shogi Explorer</name>
                            <outputDirectory>target/packages/mac</outputDirectory>
                        </configuration>
                    </execution>
                    <execution>
                        <id>bundling-for-linux</id>
                        <phase>package</phase>
                        <goals>
                            <goal>package</goal>
                        </goals>
                        <configuration>
                            <platform>linux</platform>
                            <name>ShogiExplorer</name>
                            <outputDirectory>target/packages/linux</outputDirectory>
                        </configuration>
                    </execution>
                </executions>
            </plugin>        
        </plugins>
    </build>

    <pluginRepositories>
        <pluginRepository>
            <id>nexus</id>
            <name>nexus-snapshot-repository</name>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
            <snapshots>
                <enabled>true</enabled>
                <updatePolicy>always</updatePolicy>
            </snapshots>
            <releases>
                <enabled>false</enabled>
            </releases>
        </pluginRepository>
    </pluginRepositories>

</project>
```

> Notice that common properties can be placed in `plugin.configuration` instead of `plugin.executions.execution.configuration`

**Comment by schadfield** [comment]
Snapshot 1.7.3-20230717.142757-16 does correctly include the XML file.

**Comment by fvarrui** [comment]
Great!

**Comment by fvarrui** [comment]
Branch issue-316 merged into devel, ready to be released with v1.7.3

**Comment by fvarrui** [comment]
1.7.3 released to Maven Central
